_posts/blog/2016-09-13-code-reviews.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: post
+title: "Handbook Of Manners: How To Behave In A Code Review"
+description: Some thoughts on code reviews and the way they usually are and how they should be performed.
+modified:
+categories: blog
+author: henning_gross
+excerpt: Code reviews are a common practice for knowledge distribution, learning and increasing quality - but do we perform them in a way that actually creates value?
+tags: [Mindset, Culture, Agile, XP]
+date: 2016-09-13T17:49:00-04:00
+---
+
+In most companies __code reviews__ are part of the __process__ today. In theory they help to distribute knowledge, onboard new people and improve learning and quality.
+
+
+A code review, no matter if it is performed in a pairing manner or asynchronously, is a situation in which one person __inspects__ the work of another person.
+
+
+This can lead to the __assumption__ that the reviewer has to accept the changes the author wants to introduce and put the reviewer in a position of power over the author who has to __seek approval__.
+
+
+__Agile development teams__ strive for an identity in which everyone is __equally__ important, equally __worthy__ and in which hierarchies typically make __no difference__ on functional matters (as long as the manager doesn't have to protect the business, but that's another story).
+
+
+So what does a code review ideally look like?
+
+
+Let's start with the assumption that every reviewer has the same intent - which is to __convince__ the author to follow the suggested changes (to improve the quality of the contributed code).
+
+
+When this is the main __intent__, different approaches to code reviews can only be different __strategies__ to achieve the same goal.
+
+
+The following strategies/behavioral advice can help to achieve this goal and at the same time improve the team spirit and culture.
+
+
+## Consult - Don't Preach
+
+The basic mindset in which an engineer should approach a code review for a change introduced by another engineer is the mindset of a __consultant__. Whatever you say or write; do it with caution. Sentences that start with _I would approach the problem this way..._ or _I believe this way would be more efficient_ are a good approach if you want to encourage the author to accept your recommendations. Way better than _This solution is inappropriate!_ or _Do it like this!_. Command-style comments are guaranteed to result in damage, conflicts, decreased creativity and motivation and bad team spirit. Don't do it.
```

Comment:
This is not some you need to correct or even answer, think of it a your first comment to your published blog post. 

At what point do people start ignoring the "I believe...", "I would..." prefixes? Does it not become like using "please"? Perhaps using the consultant approach is good for building an initial repertoire with the author, but if the reviewer is does not consistent apply the consultant approach through their work interaction (including those outside of reviews), then the author will likely see through such platitudes. With my colleagues with whom I think I have a particularly good repertoire, I often give fairly blunt suggestions, because (hopefully!) they know that I'm doing it with good intentions: conveying my feedback and reducing the review time by having to type fewer words (and having to spend less time thinking of a nice sentence).